src/Umbraco.Core/Strings/DefaultShortStringHelper.cs

```diff
@@ -479,11 +480,11 @@ internal string CleanCodeString(string text, CleanStringType caseType, char sepa
                     throw new Exception("Invalid state.");
             }
 
-            return new string(output, 0, opos);
+            return new string(output.Slice(0, opos));
```

Comment:
Consider using a Stringpool to convert to string if it's likely multiple calls would result in generating the same string. https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.highperformance.buffers.stringpool?view=win-comm-toolkit-dotnet-7.0